File: day02/p1.pl
#!/usr/bin/env perl
use v5.34;
my ($pos, $depth) = (0,0);
while (<>) {
  if (/forward (\d+)/) { $pos+=$1 }
  elsif (/down (\d+)/) { $depth+=$1 }
  elsif (/up (\d+)/) { $depth-=$1 }
}
say $pos * $depth;

File: 2021/day02/p1.py
#!/usr/bin/env python
import fileinput
pos, depth = 0, 0
for line in fileinput.input():
  command, value = line.split()
  match command:
    case 'forward':
      pos += int(value)
    case 'down':
      depth += int(value)
    case 'up':
      depth -= int(value)

print(pos * depth)
